MaxText/layers/mistral.py

```diff
@@ -59,47 +70,43 @@ def __call__(
       page_state=None,
       slot=None,
   ):
-    cfg = self.config
-    mesh = self.mesh
 
     inputs = nn.with_logical_constraint(inputs, ("activation_batch", "activation_norm_length", "activation_embed"))
     inputs = checkpoint_name(inputs, "decoder_layer_input")
-    lnx_rms = rms_norm(
+    lnx_rms = RMSNorm(
         num_features=inputs.shape[-1],
-        dtype=cfg.dtype,
-        weight_dtype=cfg.weight_dtype,
-        name="pre_self_attention_layer_norm",
+        dtype=self.config.dtype,
+        weight_dtype=self.config.weight_dtype,
         kernel_axes=("norm",),
-        epsilon=cfg.normalization_layer_epsilon,
+        epsilon=self.config.normalization_layer_epsilon,
```

Comment:
Looking good so far, thanks @NuojCheng. You will want to move NNX module initiation like `RMSNorm`, `Attention`, etc. to the `__init__` of `MistralDecoderLayer`